Source/WebCore/rendering/HitTestResult.cpp

```diff
@@ -528,6 +529,48 @@ void HitTestResult::enterFullscreenForVideo() const
 #endif
 }
 
+bool HitTestResult::videoHasSubtitles() const
+{
+#if ENABLE(VIDEO)
+    if (HTMLMediaElement* mediaElement = this->mediaElement()) {
+        if (auto* textTracks = mediaElement->textTracks())
+            return textTracks->length();
+    }
+#endif
+    return false;
+}
+
+bool HitTestResult::subtitlesAreShowing() const
+{
+#if ENABLE(VIDEO)
+    if (HTMLMediaElement* mediaElement = this->mediaElement()) {
+        if (mediaElement->isVideo()) {
+            if (auto* textTracks = mediaElement->textTracks(); textTracks && textTracks->length()) {
+                for (unsigned i = 0; i < textTracks->length(); i++) {
+                    if (textTracks->item(i)->mode() == TextTrack::Mode::Showing)
+                        return true;
+                }
+            }
+        }
+    }
+#endif
+    return false;
+}
+
+void HitTestResult::toggleShowSubtitlesForVideo() const
+{
+#if ENABLE(VIDEO)
+    if (HTMLMediaElement* mediaElement = this->mediaElement()) {
+        if (mediaElement->isVideo()) {
+            if (subtitlesAreShowing())
+                mediaElement->setSelectedTextTrack(&TextTrack::captionMenuOffItem());
+            else
+                mediaElement->setSelectedTextTrack(&TextTrack::captionMenuAutomaticItem());
+        }
+    }
```

Comment:
I think this would be more idiomatic with early returns and a local variable:

```
    auto* mediaElement = this->mediaElement();
    if (!mediaElement || !mediaElement->isVideo())
        return;

    auto* selectedTrack = subtitlesAreShowing() ? ...
```